ApplicationLibCode/ProjectDataModel/Summary/RimSummaryPlot.cpp

```diff
@@ -3168,14 +3242,148 @@ void RimSummaryPlot::assignPlotAxis( RimSummaryCurve* destinationCurve )
+
+                if ( !c->unitNameX().empty() ) return true;
+            }
+
+            return false;
+        };
+
+        if ( !anyCurveWithUnitText() ) strategy = AxisAssignmentStrategy::USE_MATCHING_VECTOR;
+
+        if ( strategy == AxisAssignmentStrategy::USE_MATCHING_VECTOR )
+        {
+            // Special handling if curve unit is matching. Try to match on summary vector name to avoid creation of new axis
+
+            for ( auto c : summaryCurves() )
+            {
+                if ( c == curve ) continue;
+
+                if ( c->summaryAddressX().vectorName() == curve->summaryAddressX().vectorName() )
+                {
+                    curve->setTopOrBottomAxisX( c->axisX() );
+                    return;
+                }
+            }
+        }
+        else if ( strategy == AxisAssignmentStrategy::USE_MATCHING_UNIT )
+        {
+            bool isTopUsed    = false;
+            bool isBottomUsed = false;
+
+            for ( auto c : summaryCurves() )
+            {
+                if ( c == curve ) continue;
+
+                if ( c->axisX() == RiuPlotAxis::defaultTop() ) isTopUsed = true;
+                if ( c->axisX() == RiuPlotAxis::defaultBottomForSummaryVectors() ) isBottomUsed = true;
+
+                auto currentUnit = RiaStdStringTools::toUpper( c->unitNameX() );
+
+                if ( currentUnit == destinationUnit )
+                {
+                    for ( RimPlotAxisPropertiesInterface* axisProperties : m_axisPropertiesArray )
+                    {
+                        if ( axisProperties->plotAxis().axis() == RiaDefines::PlotAxis::PLOT_AXIS_TOP ||
+                             axisProperties->plotAxis().axis() == RiaDefines::PlotAxis::PLOT_AXIS_BOTTOM )
+                        {
+                            curve->setTopOrBottomAxisX( c->axisX() );
+
+                            return;
+                        }
+                    }
+                }
+            }
+
+            if ( !isTopUsed )
+            {
+                curve->setTopOrBottomAxisX( RiuPlotAxis::defaultTop() );
+                return;
+            }
+
+            if ( !isBottomUsed )
+            {
+                curve->setTopOrBottomAxisX( RiuPlotAxis::defaultBottomForSummaryVectors() );
+                return;
+            }
+
+            strategy = AxisAssignmentStrategy::ALTERNATING;
+        }
+
+        RiaDefines::PlotAxis plotAxisType = RiaDefines::PlotAxis::PLOT_AXIS_TOP;
+
+        if ( strategy == AxisAssignmentStrategy::ALTERNATING )
+        {
+            size_t axisCountTop = 0;
+            size_t axisCountBot = 0;
+            for ( const auto& ap : m_axisPropertiesArray )
+            {
+                if ( ap->plotAxis().axis() == RiaDefines::PlotAxis::PLOT_AXIS_TOP )
+                    axisCountTop++;
+                else if ( ap->plotAxis().axis() == RiaDefines::PlotAxis::PLOT_AXIS_BOTTOM )
+                    axisCountBot++;
+            }
```

Comment:
I think something like this could work too, but not sure if better:
`size_t axisCountTop = std::count_if(array.begin(), array.end(), [](auto ap){ ap->plotAxis().axis() == RiaDefines::PlotAxis::PLOT_AXIS_TOP});`

Reply:
Fixed by using a lambda